src/rgw/rgw_lua_background.h

```diff
@@ -118,8 +118,11 @@ struct RGWTable : EmptyMetaTable {
          }
          break;
       case LUA_TSTRING:
-        value = lua_tolstring(L, 3, &len);
+      {
+        const auto str = lua_tolstring(L, 3, &len);
+        std::get<std::string>(value).assign(str, len);
```

Comment:
another way to do this is something like
```c++
value = std::string{str, len};
```

probably more readable this way, IMHO.

Reply:
the problem with `value = std::string{str};` is that it has a call to `strlen()`.
since `lua_tolstring()` already return the length, i will use:
```
value = std::string{str, len};
```